README.md

```diff
@@ -13,128 +13,145 @@ Simple cluster and environment specific aware templating for kubernetes manifest
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 **Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
 
-- [installation](#installation)
-- [usage](#usage)
-  - [scaffolding](#scaffolding)
-  - [validate files](#validate-files)
-  - [generate files](#generate-files)
+- [Installation](#installation)
+  - [Completion](#completion)
+- [Concepts](#concepts)
+  - [*Clusters* and *Environments*](#clusters-and-environments)
+  - [Templating](#templating)
+    - [Template helper functions](#template-helper-functions)
+- [Usage](#usage)
+  - [Scaffolding](#scaffolding)
+  - [Config management](#config-management)
+  - [Validate templates](#validate-templates)
+  - [Generate manifests](#generate-manifests)
   - [Overriding templates](#overriding-templates)
   - [Managing secrets](#managing-secrets)
-    - [SSM](#ssm)
+    - [Providers](#providers)
+      - [SSM](#ssm)
+      - [Random](#random)
 - [TODO](#todo)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## installation
+## Installation
 
 run this
 
-```
+```bash
 $ pip install --user --upgrade k8t
 ```
 
 **note**: k8t is not Python 2 compatible
 
-## usage
+### Completion
 
-### scaffolding
+Run the following and store the file in your distribution/OS specific spot
```

Comment:
"OS specific spot" will lose a lot of people I think. You could be more specific on what the "specific spot" is without giving the complete path for Linux, and for Mac and Windows user you could eventually do some research and add it to avoid all the other people to do it each time :)

Reply:
the issue is that this will differ from distribution do distribution.. not sure - more opinions?

Reply:
What about keeping it generic like `store the file in your completion scripts dir`?

sometimes too much detail is also bad